docs/testing/python.md

```diff
@@ -497,6 +497,61 @@ Fabric admin for default controller:
   second_ctrl = fa.new_fabric_admin.NewController(nodeId=node_id)
 ```
 
+## Automating manual steps
+
+Some test plans have manual steps that require the tester to manually change the
+state of the DUT. To run these tests in a CI environment, specific example apps
+can be built such that these manual steps can be achieved by Matter or
+named-pipe commands.
+
+In the case that all the manual steps in a test script can be achieved just
+using Matter commands, you can check if the `PICS_SDK_CI_ONLY` PICS is set to
+decide if the test script should send the required Matter commands to perform
+the manual step.
+
+```python
+self.is_ci = self.check_pics("PICS_SDK_CI_ONLY")
+```
+
+In the case that a test script requires the use of named-pipe commands to
+achieve the manual steps, you can use the `write_to_app_pipe(command)` to send
+these commands. This command requires the test class to define a `self.app_pipe`
+string value with the name of the pipe. This depends on how the app is set up.
+
+If the name of the pipe is dynamic and based on the app's PID, the following
+snippet can be added to the start of tests that use the `write_to_app_pipe`
+method.
+
+```python
+app_pid = self.matter_test_config.app_pid
+if app_pid != 0:
+	self.is_ci = true
+	self.app_pipe = "/tmp/chip_<app name>_fifo_" + str(app_pid)
```

Comment:
Let's automate that in a follow-up. This should not be duplicated everywhere and some tests actually auto-detect the PID

Reply:
Tracked using https://github.com/project-chip/connectedhomeip/issues/35284